### PR TITLE
8282473: Refactor swing classes javadoc to use @throws instead of @exception

### PR DESCRIPTION
**Pull request by prsadhuk**
Prevailing JDK coding practices use "@throws" rather than "@exception".
Refactor existing swing classes javadoc to use @throws

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8282473](https://bugs.openjdk.java.net/browse/JDK-8282473): Refactor swing classes javadoc to use @throws instead of @exception

### Reviewers
 * [Alexey Ivanov](https://openjdk.java.net/census#aivanov) (@aivanov-jdk - **Reviewer**)
 * [Dmitry Markov](https://openjdk.java.net/census#dmarkov) (@dmarkov20 - **Reviewer**)
 * [Phil Race](https://openjdk.java.net/census#prr) (@prrace - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/7644/head:pull/7644` \
`$ git checkout pull/7644`

Update a local copy of the PR: \
`$ git checkout pull/7644` \
`$ git pull https://git.openjdk.java.net/jdk pull/7644/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 7644`

View PR using the GUI difftool: \
`$ git pr show -t 7644`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/7644.diff">https://git.openjdk.java.net/jdk/pull/7644.diff</a>

</details>

### CODE REVIEW

**Review by aivanov-jdk — APPROVED** [review]

**Review by dmarkov20 — APPROVED** [review]

**Comment by aivanov-jdk** [comment]
Submitted:

- [JDK-8282539](https://bugs.openjdk.java.net/browse/JDK-8282539): Remove ending punctuation from @throws HeadlessException
- [JDK-8282546](https://bugs.openjdk.java.net/browse/JDK-8282546): Remove ending punctuation from @throws in java.desktop

I decided to separate these two cases.

File: src/java.desktop/share/classes/javax/swing/JRootPane.java [hunk 1]
```diff
@@ -578,7 +578,7 @@ public void setMenuBar(JMenuBar menu){
      * replace it with an opaque <code>JComponent</code>.
      *
      * @param content the <code>Container</code> to use for component-contents
-     * @exception java.awt.IllegalComponentStateException (a runtime
+     * @throws java.awt.IllegalComponentStateException (a runtime
      *            exception) if the content pane parameter is <code>null</code>
```
[the remark this answers is not in the document]
Comment (aivanov-jdk) [reply src/java.desktop/share/classes/javax/swing/JRootPane.java:582]: Submitted [JDK-8282562](https://bugs.openjdk.java.net/browse/JDK-8282562): Remove "(a runtime exception)" from javadoc for unchecked exceptions in java.desktop

**Review by prrace — APPROVED** [review]
This is touching a lot of files but I can't see it interfering with clean backports as we are very unlikely to touch lines in javadoc of SE API in backports !
And of course by implication, this fix is not one that would/should be backported.

**Pull request closed without merging**